File: microservices/jaxbexample/src/main/java/net/medrag/microservices/jaxbexample/controller/RController.java
package net.medrag.microservices.jaxbexample.controller;

import lombok.RequiredArgsConstructor;
import net.medrag.microservices.jaxbexample.service.SkillService;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * @author Stanislav Tretyakov
 * 31.12.2020
 */
@RestController
@RequiredArgsConstructor
public class RController {

    private final SkillService skillService;

    @GetMapping(produces = MediaType.APPLICATION_XML_VALUE)
    public String getAllSkills(){
        return skillService.getAllSkills();
    }
}

File: WsSimulator/src/main/kotlin/net/medrag/WsSimulator/controller/RController.kt
package net.medrag.WsSimulator.controller

import net.medrag.WsSimulator.model.Response
import net.medrag.WsSimulator.service.JpaService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import net.medrag.WsSimulator.shutDown


/**
 * {@author} Stanislav Tretyakov
 * 05.11.2019
 */
@RestController
@RequestMapping("/ws", "/")
class RController(private val service: JpaService) {

    @PostMapping("/addResp")
    fun addResp(@RequestBody resp: Response, @RequestParam(required = false) validate: Boolean) = try {
        ResponseEntity.ok(service.addResponse(resp, validate))
    } catch (e: Exception) {
        ResponseEntity(e.message, HttpStatus.BAD_REQUEST)
    }

    @DeleteMapping("/remove")
    fun remove(@RequestParam id: String) = ResponseEntity.ok(service.remove(id))

    @GetMapping("/getResp")
    fun getResp(@RequestParam id: String) = ResponseEntity.ok(service.getById(id))

    @GetMapping("/getReqs")
    fun getAll() = ResponseEntity.ok(service.getReqs())

    @GetMapping("/shutdown")
    fun shutdown() {
        shutDown()
    }
}